x-pack/plugins/security/common/licensing/license_service.ts

```diff
@@ -4,25 +4,43 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { Observable, Subscription, BehaviorSubject } from 'rxjs';
 import { deepFreeze } from '../../../../../src/core/utils';
-import { ILicense } from '../../../licensing/server';
+import { ILicense } from '../../../licensing/common/types';
 import { SecurityLicenseFeatures } from './license_features';
 
 export interface SecurityLicense {
+  getChanges$(): Observable<void>;
   isEnabled(): boolean;
   getFeatures(): SecurityLicenseFeatures;
 }
 
+interface SetupDeps {
+  license$: Observable<ILicense>;
+}
+
 export class SecurityLicenseService {
-  public setup() {
+  private licenseSubscription?: Subscription;
+
+  public setup({ license$ }: SetupDeps) {
     let rawLicense: Readonly<ILicense> | undefined;
 
-    return {
-      update(newRawLicense: Readonly<ILicense>) {
-        rawLicense = newRawLicense;
-      },
+    const changesSubject$ = new BehaviorSubject<void>(undefined);
+
+    this.licenseSubscription = license$.subscribe(nextRawLicense => {
+      rawLicense = nextRawLicense;
+      changesSubject$.next();
+    });
 
+    return {
       license: deepFreeze({
+        /**
+         * Observable which notifies consumers when new license information is available.
+         */
+        getChanges$() {
```

Comment:
I originally had this as a simple `changes$` property, but `deepFreeze` ends up clobbering the observable, which, depending on how I implemented the observable, either:
- caused `deepFreeze` to exhaust the call stack, presumably due to an object cycle internal to the observable
- clobber the observable in a way that caused it to never emit to subscribers

Reply:
That's sad (ping @elastic/kibana-platform in case someone else gets hit by this in the future) :cry: 

Maybe in this particular case we can just use `Object.freeze` then? Naming is the hardest :slightly_smiling_face: I like `changes$` or `change$` (like in `changeStream`), but I remember  you have similar property in a `SpacesManager` (`onActiveSpaceChange$`) - I'm open to unifying the naming between these, whatever naming pattern you like more.

Reply:
😢

Reply:
We might want to modify our `deepFreeze` utility to handles observables as non-objects? They should probably be considered as `final` properties?

Reply:
> Maybe in this particular case we can just use Object.freeze then? Naming is the hardest 🙂 I like changes$ or change$ (like in changeStream), but I remember you have similar property in a SpacesManager (onActiveSpaceChange$) - I'm open to unifying the naming between these, whatever naming pattern you like more.

Yeah for this specific case, I think Object.freeze will be sufficient

Reply:
I have an alternative proposal which exposes `features$` instead, which satisfies my current needs, and doesn't come with the mismatch between `changes$` and `BehaviorSubject`, while still being easy to consume: https://github.com/elastic/kibana/pull/52386/commits/1ac4f1a57d1644fd4fc4d8581cf003b4eb3856c5

What are everyones' thoughts about this?

Reply:
Looks good to me, let's try it and see how it goes.